examples/aggregator/module/rust/src/lib.rs

```diff
@@ -35,159 +35,142 @@ pub mod proto {
 }
 
 pub mod data;
+pub mod handler;
 
-use aggregator_common::ThresholdAggregator;
-use data::SparseVector;
+use anyhow::Context;
 use either::Either;
-use log::{debug, error, info};
+use log::{debug, info};
 use oak::{
-    grpc,
-    io::{Receiver, ReceiverExt, SenderExt},
+    io::{Receiver, ReceiverExt, Sender, SenderExt},
     proto::oak::invocation::{GrpcInvocation, GrpcInvocationSender},
 };
 use oak_abi::{label::Label, proto::oak::application::ConfigMap};
-use proto::oak::examples::aggregator::{
-    Aggregator, AggregatorClient, AggregatorDispatcher, AggregatorInit, Sample,
-};
-use std::{collections::HashMap, convert::TryFrom};
+use oak_services::proto::google::rpc;
+use proto::oak::examples::aggregator::AggregatorInit;
 
-/// Currently threshold value is hardcoded.
-pub const SAMPLE_THRESHOLD: u64 = 5;
+/// Create initialization message for Aggregator Node.
+fn create_init_message(sender: &oak::io::Sender<GrpcInvocation>) -> AggregatorInit {
+    AggregatorInit {
+        grpc_client_invocation_sender: Some(GrpcInvocationSender {
+            sender: Some(sender.clone()),
+        }),
+    }
+}
 
-/// Oak Node that collects and aggregates data.
-/// Data is collected in the `aggregators` map where keys are buckets and values are instances of a
-/// `ThresholdAggregator`. `Option` is used to keep note of the outdated buckets: once an
-/// Aggregator has sent its data to the Backend, it's replaced with `None` and all subsequent client
-/// requests corresponding to its bucket are discarded.
-pub struct AggregatorNode {
-    aggregator_client: AggregatorClient,
-    aggregators: HashMap<String, Option<ThresholdAggregator<SparseVector>>>,
+/// Returns whether the provided label is valid in the context of this aggregator application.
+fn is_valid_label(_label: &Label) -> bool {
+    true
+    // TODO(#1357): Verify that the label corresponds to exactly a single `WebAssemblyModuleTag`.
+    // (label.confidentiality_tags.len() == 1)
+    //     && (if let Some(tag) = &label.confidentiality_tags[0].tag {
+    //         match tag {
+    //             tag::Tag::GrpcTag(_) => true,
+    //             _ => false,
+    //         }
+    //     } else {
+    //         false
+    //     })
```

Comment:
Why can this not be implemented now? AFAICT it does not depend on challenge response or any missing functionality.